test/e2e/CustomComponents.test.js

```diff
@@ -0,0 +1,68 @@
+const { execute } = require("../../lib");
+const { createMockStreams, allArgs } = require("./E2ETests");
+const lolex = require("lolex");
+const brs = require("brs");
+const { RoSGNode, BrsString, Callable, ValueKind } = brs.types;
+const path = require("path");
+
+class Foo extends RoSGNode {
+    constructor(initializedFields = [], name = "Foo") {
+        super([], name);
+        this.name = name;
+
+        this.registerDefaultFields(this.defaultFields);
+        this.registerInitializedFields(initializedFields);
+
+        this.sayHello = new Callable("sayHello", {
+            signature: {
+                args: [],
+                returns: ValueKind.String,
+            },
+            impl: (_interpreter) => {
+                return new BrsString("Foo Says Hello world");
+            },
+        });
+
+        this.registerMethods({
+            ifHelloWorld: [this.sayHello],
+        });
+    }
+}
+
+describe("Extending custom components", () => {
+    let outputStreams;
+    let clock;
+
+    beforeAll(() => {
+        clock = lolex.install({ now: 1547072370937 });
+        brs.types.ComponentFactory.addComponentTypes([["Foo", (name) => new Foo([], name)]]);
+        brs.types.AddAdditionalBrsObjects([["Foo", (_interpreter) => new Foo()]]);
+
+        outputStreams = createMockStreams();
+    });
+
+    afterEach(() => {
+        jest.resetAllMocks();
+    });
+
+    afterAll(() => {
+        clock.uninstall();
+        jest.restoreAllMocks();
+    });
+
+    test("components/ExtendCustom.brs", async () => {
+        outputStreams.root = __dirname + "/customResources";
```

Comment:
Please use `path.join` so this test doesn't fail on Windows.  Windows support isn't great right now, but I'd prefer to not make it worse 😅 

```suggestion
        outputStreams.root = path.join(__dirname, "customResources");
```

Thanks for putting these resources in a separate directory though — it's extremely helpful!